ambu

```diff
@@ -1,22 +1,25 @@
 #!/bin/bash
 
 MAKEMODE=$1
-CURRENTDIR=`pwd`
+CURRENTDIR=${2:-`pwd`}
 PROJECT=`basename $CURRENTDIR`
 PARENTDIR=`dirname $CURRENTDIR`
 BUILDDIR=$PARENTDIR/".build_"$PROJECT
 PROJECTNAME=$PROJECT".ino"
 
-usage()
-{
-cat << EOF
-usage: $0 [makefile target]
-(see Arduino Makefile for target details)
-EOF
+usage() {
+    echo "usage: $0 target [project dir]"
```

Comment:
@Adman in the UNIX tradition command line arguments are usually words (i.e. separated by whitespace. This usage looks like there are three command line arguments, two of them optional. In order to fight this confusion I believe you should either connect the multiword parameter with a hyphen (`-`) or come up with a better one word parameter name.